src/hls-adapter.js

```diff
@@ -1203,6 +1217,36 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
     }
   }
 
+  /**
+   * called when a level is loaded
+   * @private
+   * @param {any} e - the event object
+   * @param {any} data - the event data
+   * @returns {void}
+   */
+  _onLevelLoaded = (e: any, data: any) => {
+    const {
+      details: {endSN}
+    } = data;
+    if (this._lastLoadedFragSN === endSN) {
+      this._sameFragSNLoadedCount++;
+      HlsAdapter._logger.debug(`Same frag SN. Count is: ${this._sameFragSNLoadedCount}, Max is: ${this._config.network.maxStaleLevelReloads}`);
+      if (this._sameFragSNLoadedCount >= this._config.network.maxStaleLevelReloads) {
+        HlsAdapter._logger.error(`Same frag loading reached max count`);
+        const error = new Error(Error.Severity.CRITICAL, Error.Category.NETWORK, Error.Code.LIVE_MANIFEST_REFRESH_ERROR, {
+          fragSN: endSN
+        });
+        this._trigger(EventType.ERROR, error);
+        return this.destroy();
+      }
+    } else {
+      HlsAdapter._logger.debug(`Different frag SN is loaded. Reset count`);
+      this._sameFragSNLoadedCount = 0;
+    }
+    HlsAdapter._logger.debug(`Last frag SN is: ${endSN}`);
```

Comment:
I'm afraid this and the log above will spam the console. they will be in any level loading.

Reply:
Although it's debug level, change to print it only if same frag is loaded.